lib/FrameManager.js

```diff
@@ -719,6 +722,20 @@ class Frame {
     return new WaitTask(this, pageFunction, 'function', polling, timeout, ...args).promise;
   }
 
+  async waitForNavigation(options = {}) {
+    const timeout = typeof options.timeout === 'number' ? options.timeout : 30000;
```

Comment:
This should use the default timeout from the page. You can just do ‘this._page._defaultTimeout.